### PR TITLE
Add branch protection for minikube

### PR DESCRIPTION
**Pull request by tstromberg**
To help prevent regressions from being merged into the minikube repository:

- Add 1 required PR review
- Add 4 required test contexts

We intend to increase the required test contexts once we feel comfortable with how the branch protection affects our workflow.

/cc @medyagh  @priyawadhwa @sharifelgamal @afbjorklund

### CODE REVIEW

**Comment by tstromberg** [comment]
Since @wojtek-t withdrew - I'm unable to find documentation on how to get this approved and merged. Can someone lend a hand?

/cc @cblecker @fejta @wojtek-t

File: config/prow/config.yaml [hunk 1]
```diff
@@ -226,6 +226,20 @@ branch-protection:
             users: []
             teams:
             - stage-bots
+        minikube:
+          enforce_admins: true
+          restrictions: # only allow admins to force-push
+            users: []
+            teams:
+            - stage-bots
```
Comment (cblecker) [inline config/prow/config.yaml:234]: As this isn't a staging repo, you can just leave this blank:
```suggestion
            teams: []
```

File: config/prow/config.yaml [hunk 2]
```diff
@@ -226,6 +226,20 @@ branch-protection:
...
+          required_status_checks:
+            contexts:
+            - functional_test_docker_ubuntu
+            - functional_baremetal_ubuntu18_04
+            - multinode_pause_tests_docker_ubuntu
+            - unit_test
+          required_pull_request_reviews:
+            required_approving_review_count: 1
```
Comment (cblecker) [inline config/prow/config.yaml:242]: As you are using OWNERS files and the approve plugin, this isn't necessary
```suggestion
```

**Comment by cblecker** [comment]
@tstromberg Once you make the above changes, I can approve this for you, no problem.

**Comment by fejta** [comment]
/unassign

LGTM

**Pull request closed without merging**